openai.py

```diff
@@ -1,185 +1,305 @@
-import os

```

Comment:
There are multiple duplicate definitions of the `validate_input` function, which can cause confusion and errors. Consolidate them into a single definition. Also, there are redundant comments in the `subtract` function docstring.

```python
+def validate_input(input_str):
+    try:
+        value = float(input_str)
+        return value
+    except ValueError:
+        raise ValueError("Invalid input! Please enter a valid number.")
```

Please remove the duplicated implementations and redundant comments for maintainability and clarity.